Content.Client/Teleportation/Ui/TeleportLocationsBoundUserInterface.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Content.Shared.Teleportation;
+using Content.Shared.Teleportation.Components;
+using JetBrains.Annotations;
+using Robust.Client.UserInterface;
+
+namespace Content.Client.Teleportation.Ui;
+
+[UsedImplicitly]
+public sealed class TeleportLocationsBoundUserInterface : BoundUserInterface
+{
+    [ViewVariables]
+    private TeleportMenu? _menu;
+
+    protected override void ReceiveMessage(BoundUserInterfaceMessage message)
+    {
+        base.ReceiveMessage(message);
+
+        if (_menu is null)
+            return;
+
+        if (message is TeleportLocationRequestCloseMessage)
+            CloseBUIResponse();
+    }
+
+    public TeleportLocationsBoundUserInterface(EntityUid owner, Enum uiKey) : base(owner, uiKey)
+    {
+    }
+
+    protected override void Open()
+    {
+        base.Open();
+
+        _menu = this.CreateWindow<TeleportMenu>();
+        SendMessage(new TeleportLocationRequestPointsMessage());
+
```

Comment:
Kid named SetUIState

Reply:
Please explain further on why this is suggested.

Reply:
You should be using UI states for stateful data not messages.